unlock-app/src/middlewares/lockMiddleware.js

```diff
@@ -98,7 +99,14 @@ export default function lockMiddleware({ getState, dispatch }) {
     dispatch(updateTransaction(transaction.hash, update))
   })
 
-  web3Service.on('error', error => {
+  web3Service.on('error', (error, transaction) => {
+    if (transaction && transaction.type === TRANSACTION_TYPES.LOCK_CREATION) {
+      // delete the lock
+      dispatch(deleteLock(transaction.lock))
+      return dispatch(
+        setError(<p>Failed to create log. Did you decline the transaction?</p>)
```

Comment:
Sending jsx to a reducer is usually an anti-pattern, because it disrupts things like the ability to time travel debug, and also to serialize state. Better is to define a set of limited possible values with some display metadata and construct the jsx when the error is rendered